api/server/creds.go

```diff
@@ -84,7 +84,7 @@ func (vd *volAPI) credsValidate(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err := d.CredsValidate(uuid); err != nil {
-		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusUnprocessableEntity)
```

Comment:
Isn't this assuming what the error is? d.CredsValidate(uuid) should return an known error, then api/server/creds can check for this error

Reply:
Currently we return an error with message which is the reason for cred validation. So for any cred validation error, I want to return 422.
500 is too generic and later we would want to send 500 for something else, hence using 422 here to indicate cred validation failure.